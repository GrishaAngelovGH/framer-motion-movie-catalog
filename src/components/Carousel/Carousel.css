.carousel .slide .legend {
  background: #fff;
  color: #000;
  font-size: 16px;
}