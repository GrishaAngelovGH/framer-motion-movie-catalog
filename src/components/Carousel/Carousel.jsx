import { Carousel } from "react-responsive-carousel"
import "react-responsive-carousel/lib/styles/carousel.min.css"

import "./Carousel.css"

const CarouselWrapper = ({ className, slides }) => (
  <Carousel className={className} useKeyboardArrows infiniteLoop showThumbs={false} >
    {
      slides.map((v, i) => (
        <div key={i}>
          <img src={v.image} />
          <div className="legend hidden md:block">
            {v.paragraphs.map((content, i) => (<p key={i}>{content}</p>))}
          </div>
        </div>
      ))
    }
  </Carousel>
)

export default CarouselWrapper