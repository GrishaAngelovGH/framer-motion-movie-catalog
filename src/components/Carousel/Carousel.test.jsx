import { render } from "@testing-library/react"

import Carousel from "./Carousel"

test("should render Carousel component", () => {
  const view = render(
    <Carousel
      slides={[{ image: "image-1.jpg", paragraphs: ["content"] }]}
    />
  )

  expect(view).toMatchSnapshot()
})