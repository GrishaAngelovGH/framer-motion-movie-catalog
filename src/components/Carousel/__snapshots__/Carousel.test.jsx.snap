// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Carousel component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel-root"
        tabindex="0"
      >
        <div
          class="carousel carousel-slider"
          style="width: 100%;"
        >
          <ul
            class="control-dots"
          >
            <li
              aria-label="slide item 1"
              class="dot selected"
              role="button"
              tabindex="0"
              value="0"
            />
          </ul>
          <button
            aria-label="previous slide / item"
            class="control-arrow control-prev control-disabled"
            type="button"
          />
          <div
            class="slider-wrapper axis-horizontal"
          >
            <ul
              class="slider animated"
              style="-webkit-transform: translate3d(-100%,0,0); transform: translate3d(-100%,0,0); transition-duration: 350ms;"
            >
              <li
                class="slide selected previous"
              >
                <div>
                  <img
                    src="image-1.jpg"
                  />
                  <div
                    class="legend hidden md:block"
                  >
                    <p>
                      content
                    </p>
                  </div>
                </div>
              </li>
              <li
                class="slide selected previous"
              >
                <div>
                  <img
                    src="image-1.jpg"
                  />
                  <div
                    class="legend hidden md:block"
                  >
                    <p>
                      content
                    </p>
                  </div>
                </div>
              </li>
            </ul>
          </div>
          <button
            aria-label="next slide / item"
            class="control-arrow control-next control-disabled"
            type="button"
          />
          <p
            class="carousel-status"
          >
            1 of 1
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel-root"
      tabindex="0"
    >
      <div
        class="carousel carousel-slider"
        style="width: 100%;"
      >
        <ul
          class="control-dots"
        >
          <li
            aria-label="slide item 1"
            class="dot selected"
            role="button"
            tabindex="0"
            value="0"
          />
        </ul>
        <button
          aria-label="previous slide / item"
          class="control-arrow control-prev control-disabled"
          type="button"
        />
        <div
          class="slider-wrapper axis-horizontal"
        >
          <ul
            class="slider animated"
            style="-webkit-transform: translate3d(-100%,0,0); transform: translate3d(-100%,0,0); transition-duration: 350ms;"
          >
            <li
              class="slide selected previous"
            >
              <div>
                <img
                  src="image-1.jpg"
                />
                <div
                  class="legend hidden md:block"
                >
                  <p>
                    content
                  </p>
                </div>
              </div>
            </li>
            <li
              class="slide selected previous"
            >
              <div>
                <img
                  src="image-1.jpg"
                />
                <div
                  class="legend hidden md:block"
                >
                  <p>
                    content
                  </p>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next slide / item"
          class="control-arrow control-next control-disabled"
          type="button"
        />
        <p
          class="carousel-status"
        >
          1 of 1
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
